rero_ils/modules/documents/dojson/contrib/jsontomarc21/model.py

```diff
@@ -62,15 +61,142 @@ def set_value(data, old_data, key):
                         'qualifier', 'date_of_birth', 'date_of_death',
                         'subordinate_unit', 'conference', 'conference_number',
                         'conference_date', 'conference_place']:
-                agent = set_value(source_data, agent, key)
-            agent.pop(source)
-    agent['refs'] = refs
-    if 'bf:Agent' in agent:
-        agent['type'] = agent.pop('bf:Agent')
-    contribution['entity'] = agent
+                entity = set_value(source_data, entity, key)
+            entity.pop(source)
+    entity['refs'] = refs
+    if 'bf:Agent' in entity:
+        entity['type'] = entity.pop('bf:Agent')
+    contribution['entity'] = entity
     return contribution
 
 
+def replace_concept_sources(concept, source_order):
+    """Prepare concept data from sources.
+
+    :param concept: concept to use.
+    :source_order: Source order to use to get the informations.
+    :returns: concept entity with localized values.
+    """
+    def set_value(data, old_data, key):
+        """Set new data value."""
+        value = data.get(key)
+        if not old_data.get(key) and value:
+            old_data[key] = value
+        return old_data
+
+    refs = []
+    for source in source_order:
+        source_data = concept.get(source, {})
+        if source_data:
+            refs.append({
+                'source': source,
+                'pid': source_data.get("pid")
+            })
+            for key in ['type', 'authorized_access_point']:
+                concept = set_value(source_data, concept, key)
+            concept.pop(source)
+    concept['refs'] = refs
+    return concept
+
+
+def do_contribution(contribution, source_order):
+    """Create contribution.
+
+    :param contribution: contribution to transform.
+    :param source_order: source_order to use.
+    :returns: result marc dictionary,
```

Comment:
could you be more explicit ? reading the docstring, I don't understand the first element of the returned tuple.
Maybe a good idea to add a `:rtype: tuple(dict, str, bool, bool)`